spanner/integration_test.go

```diff
@@ -1699,6 +1718,22 @@ func TestIntegration_BasicTypes(t *testing.T) {
 		{col: "NumericArray", val: []NullNumeric{{n1, true}, {n2, true}, {}}},
 	}
 
+	if !isEmulatorEnvSet() {
+		tests = append(tests, []struct {
+			col  string
+			val  interface{}
+			want interface{}
+		}{
+			{col: "JSON", val: NullJSON{msg, true}, want: msg},
+			{col: "JSON", val: NullJSON{msg, true}, want: NullJSON{unmarshalledJSONstruct, true}},
```

Comment:
This actually surprised / confused me the first time I read it, and I thought it was not correct, but it probably is. This means that we do support **de**coding `JSON` values to Go `Struct` types, but we do **not** support **en**coding Go `Struct` types to `JSON`. That might be worth a comment in the docs.

Reply:
Fixed. To avoid confusion, I removed the code by only allowing to decode a Cloud Spanner JSON to NullJSON.